### ISSUE TITLE
Unable to revoke EC certificates

### ISSUE THREAD

**Issue by ghost** [certbot/certbot#3434]
i created both EC and RSA certs

```
cd /etc/letsencrypt/live/example.com
ls -1
    cert.ec.crt.pem
    cert.rsa.crt.pem
    chain.ec.crt.pem
    chain.rsa.crt.pem
    fullchain.ec.crt.pem
    fullchain.rsa.crt.pem
```

revoking rsa cert appears to work,

```
CA="https://acme-v01.api.letsencrypt.org/directory"
certbot-auto revoke \
--server ${CA} \
--key-path ./privkey_rsa.pem \
--cert-path ./cert.rsa.crt.pem \
-d example.com
```

since if I try again

```
...
An unexpected error occurred:
The request message was malformed :: Certificate already revoked
Please see the logfiles in /var/log/letsencrypt for more details.
```

now if I try to revoke the ec cert

```
certbot-auto revoke \
--server ${CA} \
--key-path ./privkey_ec.pem \
--cert-path ./cert.ec.crt.pem \
-d example.com
```

it fails with the

```
An unexpected error occurred:
TypeError: __init__() takes exactly the following arguments:  (key given)
Please see the logfiles in /var/log/letsencrypt for more details.
```

error

**Comment by bmw** [comment]
Creating EC certificates with Certbot is still under development (see #2163). With this in mind, it doesn't surprise me it doesn't work for revocation either as Certbot doesn't know how to parse the key.

This might get fixed along with #2163, but I'll add it to our 1.0 milestone to make sure.

**Comment by bmw** [comment]
Certbot now supports EC keys so this should be fixed.

If you're still having trouble with an up-to-date version of Certbot, please comment here or open a new issue.
